src/main/java/org/commcare/formplayer/services/NewFormResponseFactory.java

```diff
@@ -106,6 +105,10 @@ public NewFormResponse getResponse(NewSessionRequestBean bean, String postUrl) t
                 caseSearchHelper
         );
 
+        // can only setup local storage once a form session has been created
+        this.storageFactory.configure(formSession.getSerializableSession());
```

Comment:
Totally missed that. I think I got excited cause I tested this locally and it worked, but in hindsight, in order to test sms locally I need to run a menu session first so that localization is properly setup (should be able to put out a fix for that), and I'm wondering if that is the same case here. Is it possible performing an app install for a menu session prior to an sms form session allows the sms form session to have access to a sqlite db?

Reply:
I don't think that would work for SMS since it uses a different data directory so a previous app install wouldn't be accessible

Reply:
For normal sessions a previous app install of the same app for that user would have created the DBs (though each request still needs to register them with the storage manager) and it's quite possible this is only happening after we load the form def from the DB.

Reply:
@gherceg It's quite likely that a previous app install is resulting in a thread later having a Localizer. In Spring the worker threads don't terminate after completion (they are reused between requests). If the issue here is simply that a Localizer was never created on the SMS thread, a previously successful menusession request that created a threadstatic localizer would likely change the behavior for a subsequent sms session request.

Reply:
Yeah I figured the localizer mechanism and the app db mechanisms are quite different. I'm fairly confident what you are describing is exactly what happens in the case of the localizer.